metric_learn/lsml.py

```diff
@@ -66,14 +92,18 @@ def _fit(self, quadruplets, weights=None):
     else:
       self.w_ = weights
     self.w_ /= self.w_.sum()  # weights must sum to 1
-    if self.prior is None:
-      X = np.vstack({tuple(row) for row in
-                     quadruplets.reshape(-1, quadruplets.shape[2])})
-      prior_inv = np.atleast_2d(np.cov(X, rowvar=False))
-      M = np.linalg.inv(prior_inv)
-    else:
-      M = self.prior
-      prior_inv = np.linalg.inv(self.prior)
+    # if the prior is the default (identity), we raise a warning just in case
+    if self.prior == 'identity':
+      msg = ("Warning, as of version 0.5.0, the default prior is now "
+             "'identity', instead of 'covariance'. If you still want to use "
+             "the inverse of the covariance matrix as a prior, "
+             "set 'prior'=='covariance'. This warning will disappear in "
+             "v0.6.0.")
+      warnings.warn(msg, ChangedBehaviorWarning)
```

Comment:
This warning will be annoying for users that want to use identity initialization intentionally. We could instead keep using `None` as the default and only warn when the user hasn't set the prior explicitly.

Reply:
I agree, let me know what you think of the new error messages (says that init was not set because is None, and that in the next version it will be set to 'identity' for instance, and that the warning will disappear)

Reply:
Yes, this is good. One small nit: instead of "set 'prior'=='covariance'.", we should do "set `prior='covariance'`."

Reply:
Agreed, will do

Reply:
done